proto/document/v1/document.proto

```diff
@@ -0,0 +1,126 @@
+syntax = "proto3";
+package nitric.document.v1;
+
+import "google/protobuf/struct.proto";
+
+// protoc plugin options for code generation
+option go_package = "nitric/v1;v1";
+option java_package = "io.nitric.proto.document.v1";
+option java_multiple_files = true;
+option java_outer_classname = "Documents";
+option php_namespace = "Nitric\\Proto\\Document\\V1";
+option csharp_namespace = "Nitric.Proto.Document.v1";
+
+// Service for storage and retrieval of simple JSON keyValue
+service Document {
+  // Get an existing document
+  rpc Get (DocGetRequest) returns (DocGetResponse);
+
+  // Create a new or overwrite an existing document
+  rpc Set (DocSetRequest) returns (DocSetResponse);
+
+  // Delete an existing document
+  rpc Delete (DocDeleteRequest) returns (DocDeleteResponse);
+  
+  // Query the document collection (pagination support)
+  rpc Query (DocQueryRequest) returns (DocQueryResponse);
+  
+  // Query All the document collection (streaming support)
+  rpc QueryAll (DocQueryAllRequest) returns (stream DocQueryAllResponse);
```

Comment:
I think <op>All is fine, would be good to look at some conventions to see how they separate out the different permutations of request/response types for gRPC though:
 - UnaryUnary
 - UnaryStream
 - StreamUnary
 - StreamStream

Reply:
QueryAll -> QueryStream could work. I wouldn't use it in an SDK API however, as this would be wrapped in some other language feature.

Reply:
I think I'd prefer QueryStream or have Query be the method that returns a steam and rename the current Query to PaginatedQuery or something like that.